Issue/2983 preserve local changes

nbradbury opened the pull request:
Fixes #2983. Right now the post list doesn't automatically refresh if there are posts with local changes. If the user does a pull-to-refresh, they're given a choice between keeping local changes or retrieving new posts. 

This PR resolves this by removing both of those limitations. Whenever the post list refreshes, regardless of whether it was automatic or due to a PTR, local changes are retained. 

This does mean that if the user edits a post outside of the app, the app won't show those edits if it has local changes. This isn't ideal, but I think is still far better than the current situation.

Note that #2997 will improve this by enabling the user to revert local changes for a post.

daniloercoli commented on WordPress/src/main/java/org/wordpress/android/WordPressDB.java:
```diff
@@ -1218,28 +1253,39 @@ private void putPostLocation(Post post, ContentValues values) {
 
-        c.close();
-        return post;
+    public Post getPostForRemotePostId(int blogID, String remotePostID) {
```
This is never used.

daniloercoli commented:
This PR looks good. It introduces a new method that is never used in the code, maybe is it something you're planning to use later?

nbradbury commented:
> It introduces a new method that is never used in the code, maybe is it something you're planning to use later?

Oh, right. I added that for this branch, then stopped using it. I just dropped it in 93ac101.

daniloercoli commented:
:shipit:
